server/monitor-types/tls.js

```diff
@@ -0,0 +1,266 @@
+
+        const timeoutMs = options.interval * 1000 * 0.8;
+        const timeoutID = setTimeout(() => {
+            this.log(`timeout after ${timeoutMs} ms`);
+            abortController.abort();
+        }, timeoutMs);
+
+        const tlsSocket = await this.connect(abortController.signal, options.hostname, options.port, options.useStartTls);
+        let tlsSocketClosed = false;
+        tlsSocket.on("close", () => {
+            tlsSocketClosed = true;
+        });
+
+        const result = await this.getResponseFromTlsPort(abortController.signal, tlsSocket, options)
+            .then((response) => {
+                clearTimeout(timeoutID);
+                return response;
+            })
+            .catch((error) => {
+                clearTimeout(timeoutID);
+                throw error;
+            })
+        ;
+
+        if (!tlsSocketClosed) {
+            tlsSocket.end();
+            tlsSocket.destroy();
+            this.debug_log("TLS socket commanded to close");
+        }
+
+        this.processResponse(result, monitor, heartbeat);
+    }
+
+    /**
+     * Compares the server's response against the monitor's attributes, decides on success/failure,
+     * and updates the heartbeat accordingly.
+     * @param {string}    response  Response received from the server
+     * @param {Monitor}   monitor   Monitor to check
+     * @param {Heartbeat} heartbeat Monitor heartbeat to update
+     * @returns {void}
+     * @throws Error if the check fails
+     */
+    processResponse(response, monitor, heartbeat) {
+        if (response instanceof Error) {
+            this.log("check failed: " + response.message);
+            throw response;
+        } else {
+            let success = false;
+            let message = undefined;
+            if (monitor.keyword) {
+                const keywordContained = response.includes(monitor.keyword);
+                success = (keywordContained === !monitor.invertKeyword);
+                message = keywordContained ? "Data contains keyword" : "Data does not contain keyword";
+            } else {
+                success = true;
+                message = "Connection successful";
+            }
+
+            if (success) {
+                this.log("check successful: " + message);
+                heartbeat.msg = message;
+                heartbeat.status = UP;
+            } else {
+                this.log("check failed: " + message);
+                throw new Error(message);
+            }
+        }
+    }
+
+    /**
+     * Sends the request over the given TLS socket and returns the response.
+     * @param {AbortController} aborter   Abort controller used to abort the request
+     * @param {TLSSocket}       tlsSocket TLS socket instance
+     * @param {*}               options   Monitor options
+     * @returns {Promise<string>} Server response on success or rejected promise on error
+     */
+    async getResponseFromTlsPort(aborter, tlsSocket, options) {
```

Comment:
please inline `request` instead, as this is the only usage of `options`

```suggestion
    async getResponseFromTlsPort(aborter, tlsSocket, request) {
```